### PR TITLE
feat: add an endpoint to clean diagnosis keys

### LINKED ISSUE #331: Endpoint for Clearing Diagnosis Keys in Database
**PLEASE NOTE:** This would be used in the staging environment only for QA Purposes only

This end point when called with an API Key should trunc the diagnosis_keys table

### PR DESCRIPTION
**Pull request by CalvinRodo**
Add a new test-tools servlet
Add tests for test-tools servlet
Add test-tools servlet to retrieve service

Refactor some existing tests in the server namespace to use the new testhelpers logging functions
Fix an issue with how the event logger test was working

Write tests for the test_tools servlet
Write tests for the test_tools database code

## How to Test

Start a new monolith server ensure the env variable ENABLE_TEST_TOOLS is true 
Add some keys to the diagnosis_keys table 
Post to /clear-diagnosis-keys and pass a bearer token
Validate the diagnosis_keys table is empty 

This closes #331

### CODE REVIEW

File: pkg/persistence/test_tools.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+func clearDiagnosisKeys(ctx context.Context, db *sql.DB) error {
+	// This should never happen but I want to be extra sure this isn't in production
+	if os.Getenv("ENABLE_TEST_TOOLS") != "true" {
+		panic("un-allowed attempt to call clearDiagnosis keys")
+	}
+
+	if _, err := db.Exec(`TRUNCATE TABLE diagnosis_keys`); err != nil {
+		return err
+	}
+	log(ctx,nil).Info("diagnosis_keys was truncated")
```
Comment (maxneuvians) [inline pkg/persistence/test_tools.go:23]: Might be worth while to log an IP here

File: pkg/testhelpers/logging.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"github.com/Shopify/goose/logger"
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func SetupTestLogging(log *logger.Logger) (*test.Hook, *logger.Logger) {
```
Comment (maxneuvians) [inline pkg/testhelpers/logging.go:11]: Nice

File: pkg/config/config.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ var log = logger.New("config")
 type Constants struct {
 	DefaultSubmissionServerPort        uint32
 	DefaultRetrievalServerPort         uint32
+	DefaultTestToolsServerPort         uint32
```
Comment (maxneuvians) [inline pkg/config/config.go:15]: Maybe remove

File: pkg/config/config.go [hunk 4]
```diff
@@ -51,6 +52,7 @@ func InitConfig() {
 func setDefaults() {
 	viper.SetDefault("defaultSubmissionServerPort", 8000)
 	viper.SetDefault("defaultRetrievalServerPort", 8001)
+	viper.SetDefault("defaultTestToolsServerPort", 8002)
```
Comment (maxneuvians) [inline pkg/config/config.go:55]: Maybe remove

File: pkg/server/test_tools.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	return &adminToolsServlet{db: db, auth: auth}
+}
+
+func (a adminToolsServlet) RegisterRouting(r *mux.Router) {
+
+	// This should never happen but I want to be extra sure this isn't in production
+	if os.Getenv("ENABLE_TEST_TOOLS") != "true" {
+		panic("test tools must be enabled to register these routes")
+	}
+
```
Comment (maxneuvians) [inline pkg/server/test_tools.go:25]: if `ENV == production`, panic

**Review by maxneuvians — APPROVED** [review]
LGTM!

File: pkg/config/config.go (hunk 3 above)
Comment (CalvinRodo) [reply pkg/config/config.go:15]: Good idea

**Pull request merged**